server.go

```diff
@@ -248,6 +249,22 @@ func (s *Server) sendResponse(t transport.ServerTransport, stream *transport.Str
 }
 
 func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.Stream, srv *service, md *MethodDesc) {
+	var (
+		traceInfo traceInfo
+		err       error
+	)
+	if EnableTracing {
+		traceInfo.tr = trace.New("Recv."+methodFamily(stream.Method()), stream.Method())
+		defer traceInfo.tr.Finish()
+		traceInfo.firstLine.client = false
+		traceInfo.tr.LazyLog(&traceInfo.firstLine, false)
+		defer func() {
+			if err != nil {
+				traceInfo.tr.LazyLog(&fmtStringer{"%v", []interface{}{err}}, true)
+				traceInfo.tr.SetError()
+			}
+		}()
+	}
 	p := &parser{s: stream}
 	for {
 		pf, req, err := p.recvMsg()
```

Comment:
@dsymonds 

FYI, my comment was before we decided to add a return value (error) for this function. github is bad on showing this ...